app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
@stanhu , thanks. I am using the 'class: "dropzone"' in file_tag of _upload.html.haml, like:

= form_tag namespace_project_create_blob_path(@project.namespace, @project, @id), method: :post, multipart: true, class: "dropzone" do
          %br
          .form-group
            .col-sm-offset-2.col-sm-10
              = file_field_tag :file_upload, required: true

I use params[:file] in the blob_controller, but it is a nil. I think I misunderstand some things here, could you give me some information? thank you very much.